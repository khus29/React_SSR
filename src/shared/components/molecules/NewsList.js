import React, { Component } from 'react';
import queryString from 'query-string';
import styled from 'styled-components';
import { fetchNewsData } from '../../api';

import List from '../atoms/List';
import { Link } from 'react-router-dom';
import ListChild from '../atoms/ListChild';
import ListItem from '../molecules/ListItem';

import { setLocalstorage, getLocalstorage } from '../../helpers';

const Div = styled.div`
  background-color: #ff4500;
  margin-left: 40px;
  padding: 10px 20px;
`;

class NewsList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 0,
      newsData: props.data,
    };
  }
  handlePageChange(pageNum) {
    fetchNewsData(pageNum).then((data) => {
      this.setState({
        currentPage: pageNum,
        newsData: data,
      });
    });
  }
  hideNews(e) {
    const parentDiv = e.target.parentElement;
    const objectId = parentDiv.getAttribute('id');
    parentDiv.parentElement && parentDiv.parentElement.classList.add('hide');

    setLocalstorage('objectId', objectId);
  }

  render() {
    const { newsData, currentPage } = this.state;

    const url = typeof window !== 'undefined' && window.location.search;
    const values = queryString.parse(url);
    const nextPage = +currentPage + 1;
    if (values.page && currentPage !== values.page) {
      this.handlePageChange(values.page);
    }

    return (
      <div>
        <p>Page number : {currentPage}</p>
        <Div>top | new</Div>
        <List>
          <React.Fragment>
            {newsData &&
              newsData.map((item) => {
                return (
                  <ListItem
                    key={item.objectID}
                    item={item}
                    hideNewsHandler={this.hideNews}
                  />
                );
              })}
            <ListChild className='pagination'>
              <Link
                to={`?page=${nextPage}`}
                className='more-link'
                aria-label='click here for more news'
              >
                More News
              </Link>
            </ListChild>
          </React.Fragment>
        </List>
      </div>
    );
  }
}

export default NewsList;
