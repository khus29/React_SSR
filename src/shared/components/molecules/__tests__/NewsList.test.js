import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import { shallow, configure, mount } from 'enzyme';
import NewsList from '../NewsList';

configure({ adapter: new Adapter() });
describe('News Component', () => {
  const props = {
    data: [
      {
        created_at: '2019-02-28T11:38:16.000Z',
        title:
          'Mozilla releases the largest to-date public domain transcribed voice dataset',
        url:
          'https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/',
        author: 'Vinnl',
        points: 705,
        comment_text: null,
        num_comments: 61,
        created_at_i: 1551353896,
        objectID: '19270646',
      },
      {
        created_at: '2019-02-28T11:38:16.000Z',
        title:
          'Mozilla releases the largest to-date public domain transcribed voice dataseta',
        url:
          'https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/',
        author: 'Vinnla',
        points: 705,
        comment_text: null,
        num_comments: 61,
        created_at_i: 1551353896,
        objectID: '19270686',
      },
    ],
  };

  test('should render correctly and match snapshot', () => {
    const tree = shallow(<NewsList {...props} />);
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly and match snapshot', () => {
    const tree = mount(
      <BrowserRouter>
        <NewsList {...props} />
      </BrowserRouter>
    );
    tree.find('Link').first();

    const nextLink = tree.find('Link').first();
    expect(nextLink.length).toBe(1);
    nextLink.simulate('click');
  });
});
