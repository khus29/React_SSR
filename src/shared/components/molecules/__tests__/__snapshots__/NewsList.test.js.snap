// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Component should render correctly and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsList
    data={
      Array [
        Object {
          "author": "Vinnl",
          "comment_text": null,
          "created_at": "2019-02-28T11:38:16.000Z",
          "created_at_i": 1551353896,
          "num_comments": 61,
          "objectID": "19270646",
          "points": 705,
          "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
          "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
        },
        Object {
          "author": "Vinnla",
          "comment_text": null,
          "created_at": "2019-02-28T11:38:16.000Z",
          "created_at_i": 1551353896,
          "num_comments": 61,
          "objectID": "19270686",
          "points": 705,
          "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
          "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(styled.div)>
          Page number : 
          1
        </ForwardRef(styled.div)>,
        <ForwardRef(styled.div)
          className="navigation"
        >
          <Anchor
            href="/"
          >
            <ForwardRef(styled.img)
              alt="news logo image"
              height="15"
              src="https://news.ycombinator.com/y18.gif"
              width="15"
            />
          </Anchor>
          <Span
            className="top-filter"
          >
            top
          </Span>
           | new
        </ForwardRef(styled.div)>,
        <List>
          <React.Fragment>
            <ListItem
              hideNewsHandler={[Function]}
              item={
                Object {
                  "author": "Vinnl",
                  "comment_text": null,
                  "created_at": "2019-02-28T11:38:16.000Z",
                  "created_at_i": 1551353896,
                  "num_comments": 61,
                  "objectID": "19270646",
                  "points": 705,
                  "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                  "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                }
              }
            />
            <ListItem
              hideNewsHandler={[Function]}
              item={
                Object {
                  "author": "Vinnla",
                  "comment_text": null,
                  "created_at": "2019-02-28T11:38:16.000Z",
                  "created_at_i": 1551353896,
                  "num_comments": 61,
                  "objectID": "19270686",
                  "points": 705,
                  "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                  "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                }
              }
            />
            <ListChild
              className="pagination"
            >
              <Link
                aria-label="click here for more news"
                className="more-link"
                replace={false}
                to="?page=1"
              >
                More News
              </Link>
            </ListChild>
          </React.Fragment>
        </List>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Page number : ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Page number : ",
          1,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "baseHash": 1105857511,
            "componentId": "sc-AxgMl",
            "isStatic": false,
            "rules": Array [
              "
  margin-bottom: 20px;
  @media (min-width: 1024px) {
    margin: 0 20px 20px;
  }

  &.navigation {
    background-color: #ff4500;
    margin: 0;
    padding: 10px 20px;
    @media (min-width: 1024px) {
      margin: 0 20px;
      padding: 10px 20px;
    }
  }
",
            ],
            "staticRulesId": "",
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-AxgMl",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Anchor
              href="/"
            >
              <ForwardRef(styled.img)
                alt="news logo image"
                height="15"
                src="https://news.ycombinator.com/y18.gif"
                width="15"
              />
            </Anchor>,
            <Span
              className="top-filter"
            >
              top
            </Span>,
            " | new",
          ],
          "className": "navigation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef(styled.img)
                alt="news logo image"
                height="15"
                src="https://news.ycombinator.com/y18.gif"
                width="15"
              />,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": "news logo image",
                "height": "15",
                "src": "https://news.ycombinator.com/y18.gif",
                "width": "15",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "baseHash": 1882630949,
                  "componentId": "sc-AxheI",
                  "isStatic": false,
                  "rules": Array [
                    "
  border: 1px #ffffff solid;
",
                  ],
                  "staticRulesId": "",
                },
                "displayName": "styled.img",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-AxheI",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "top",
              "className": "top-filter",
            },
            "ref": null,
            "rendered": "top",
            "type": [Function],
          },
          " | new",
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "baseHash": 1105857511,
            "componentId": "sc-AxgMl",
            "isStatic": false,
            "rules": Array [
              "
  margin-bottom: 20px;
  @media (min-width: 1024px) {
    margin: 0 20px 20px;
  }

  &.navigation {
    background-color: #ff4500;
    margin: 0;
    padding: 10px 20px;
    @media (min-width: 1024px) {
      margin: 0 20px;
      padding: 10px 20px;
    }
  }
",
            ],
            "staticRulesId": "",
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-AxgMl",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <React.Fragment>
            <ListItem
              hideNewsHandler={[Function]}
              item={
                Object {
                  "author": "Vinnl",
                  "comment_text": null,
                  "created_at": "2019-02-28T11:38:16.000Z",
                  "created_at_i": 1551353896,
                  "num_comments": 61,
                  "objectID": "19270646",
                  "points": 705,
                  "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                  "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                }
              }
            />
            <ListItem
              hideNewsHandler={[Function]}
              item={
                Object {
                  "author": "Vinnla",
                  "comment_text": null,
                  "created_at": "2019-02-28T11:38:16.000Z",
                  "created_at_i": 1551353896,
                  "num_comments": 61,
                  "objectID": "19270686",
                  "points": 705,
                  "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                  "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                }
              }
            />
            <ListChild
              className="pagination"
            >
              <Link
                aria-label="click here for more news"
                className="more-link"
                replace={false}
                to="?page=1"
              >
                More News
              </Link>
            </ListChild>
          </React.Fragment>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              Array [
                <ListItem
                  hideNewsHandler={[Function]}
                  item={
                    Object {
                      "author": "Vinnl",
                      "comment_text": null,
                      "created_at": "2019-02-28T11:38:16.000Z",
                      "created_at_i": 1551353896,
                      "num_comments": 61,
                      "objectID": "19270646",
                      "points": 705,
                      "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                      "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                    }
                  }
                />,
                <ListItem
                  hideNewsHandler={[Function]}
                  item={
                    Object {
                      "author": "Vinnla",
                      "comment_text": null,
                      "created_at": "2019-02-28T11:38:16.000Z",
                      "created_at_i": 1551353896,
                      "num_comments": 61,
                      "objectID": "19270686",
                      "points": 705,
                      "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                      "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                    }
                  }
                />,
              ],
              <ListChild
                className="pagination"
              >
                <Link
                  aria-label="click here for more news"
                  className="more-link"
                  replace={false}
                  to="?page=1"
                >
                  More News
                </Link>
              </ListChild>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "19270646",
              "nodeType": "class",
              "props": Object {
                "hideNewsHandler": [Function],
                "item": Object {
                  "author": "Vinnl",
                  "comment_text": null,
                  "created_at": "2019-02-28T11:38:16.000Z",
                  "created_at_i": 1551353896,
                  "num_comments": 61,
                  "objectID": "19270646",
                  "points": 705,
                  "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                  "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "19270686",
              "nodeType": "class",
              "props": Object {
                "hideNewsHandler": [Function],
                "item": Object {
                  "author": "Vinnla",
                  "comment_text": null,
                  "created_at": "2019-02-28T11:38:16.000Z",
                  "created_at_i": 1551353896,
                  "num_comments": 61,
                  "objectID": "19270686",
                  "points": 705,
                  "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                  "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  aria-label="click here for more news"
                  className="more-link"
                  replace={false}
                  to="?page=1"
                >
                  More News
                </Link>,
                "className": "pagination",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "click here for more news",
                  "children": "More News",
                  "className": "more-link",
                  "replace": false,
                  "to": "?page=1",
                },
                "ref": null,
                "rendered": "More News",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(styled.div)>
            Page number : 
            1
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)
            className="navigation"
          >
            <Anchor
              href="/"
            >
              <ForwardRef(styled.img)
                alt="news logo image"
                height="15"
                src="https://news.ycombinator.com/y18.gif"
                width="15"
              />
            </Anchor>
            <Span
              className="top-filter"
            >
              top
            </Span>
             | new
          </ForwardRef(styled.div)>,
          <List>
            <React.Fragment>
              <ListItem
                hideNewsHandler={[Function]}
                item={
                  Object {
                    "author": "Vinnl",
                    "comment_text": null,
                    "created_at": "2019-02-28T11:38:16.000Z",
                    "created_at_i": 1551353896,
                    "num_comments": 61,
                    "objectID": "19270646",
                    "points": 705,
                    "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                    "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                  }
                }
              />
              <ListItem
                hideNewsHandler={[Function]}
                item={
                  Object {
                    "author": "Vinnla",
                    "comment_text": null,
                    "created_at": "2019-02-28T11:38:16.000Z",
                    "created_at_i": 1551353896,
                    "num_comments": 61,
                    "objectID": "19270686",
                    "points": 705,
                    "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                    "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                  }
                }
              />
              <ListChild
                className="pagination"
              >
                <Link
                  aria-label="click here for more news"
                  className="more-link"
                  replace={false}
                  to="?page=1"
                >
                  More News
                </Link>
              </ListChild>
            </React.Fragment>
          </List>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Page number : ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Page number : ",
            1,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "baseHash": 1105857511,
              "componentId": "sc-AxgMl",
              "isStatic": false,
              "rules": Array [
                "
  margin-bottom: 20px;
  @media (min-width: 1024px) {
    margin: 0 20px 20px;
  }

  &.navigation {
    background-color: #ff4500;
    margin: 0;
    padding: 10px 20px;
    @media (min-width: 1024px) {
      margin: 0 20px;
      padding: 10px 20px;
    }
  }
",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-AxgMl",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Anchor
                href="/"
              >
                <ForwardRef(styled.img)
                  alt="news logo image"
                  height="15"
                  src="https://news.ycombinator.com/y18.gif"
                  width="15"
                />
              </Anchor>,
              <Span
                className="top-filter"
              >
                top
              </Span>,
              " | new",
            ],
            "className": "navigation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef(styled.img)
                  alt="news logo image"
                  height="15"
                  src="https://news.ycombinator.com/y18.gif"
                  width="15"
                />,
                "href": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "alt": "news logo image",
                  "height": "15",
                  "src": "https://news.ycombinator.com/y18.gif",
                  "width": "15",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "baseHash": 1882630949,
                    "componentId": "sc-AxheI",
                    "isStatic": false,
                    "rules": Array [
                      "
  border: 1px #ffffff solid;
",
                    ],
                    "staticRulesId": "",
                  },
                  "displayName": "styled.img",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-AxheI",
                  "target": "img",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "top",
                "className": "top-filter",
              },
              "ref": null,
              "rendered": "top",
              "type": [Function],
            },
            " | new",
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "baseHash": 1105857511,
              "componentId": "sc-AxgMl",
              "isStatic": false,
              "rules": Array [
                "
  margin-bottom: 20px;
  @media (min-width: 1024px) {
    margin: 0 20px 20px;
  }

  &.navigation {
    background-color: #ff4500;
    margin: 0;
    padding: 10px 20px;
    @media (min-width: 1024px) {
      margin: 0 20px;
      padding: 10px 20px;
    }
  }
",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-AxgMl",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <React.Fragment>
              <ListItem
                hideNewsHandler={[Function]}
                item={
                  Object {
                    "author": "Vinnl",
                    "comment_text": null,
                    "created_at": "2019-02-28T11:38:16.000Z",
                    "created_at_i": 1551353896,
                    "num_comments": 61,
                    "objectID": "19270646",
                    "points": 705,
                    "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                    "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                  }
                }
              />
              <ListItem
                hideNewsHandler={[Function]}
                item={
                  Object {
                    "author": "Vinnla",
                    "comment_text": null,
                    "created_at": "2019-02-28T11:38:16.000Z",
                    "created_at_i": 1551353896,
                    "num_comments": 61,
                    "objectID": "19270686",
                    "points": 705,
                    "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                    "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                  }
                }
              />
              <ListChild
                className="pagination"
              >
                <Link
                  aria-label="click here for more news"
                  className="more-link"
                  replace={false}
                  to="?page=1"
                >
                  More News
                </Link>
              </ListChild>
            </React.Fragment>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                Array [
                  <ListItem
                    hideNewsHandler={[Function]}
                    item={
                      Object {
                        "author": "Vinnl",
                        "comment_text": null,
                        "created_at": "2019-02-28T11:38:16.000Z",
                        "created_at_i": 1551353896,
                        "num_comments": 61,
                        "objectID": "19270646",
                        "points": 705,
                        "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                        "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                      }
                    }
                  />,
                  <ListItem
                    hideNewsHandler={[Function]}
                    item={
                      Object {
                        "author": "Vinnla",
                        "comment_text": null,
                        "created_at": "2019-02-28T11:38:16.000Z",
                        "created_at_i": 1551353896,
                        "num_comments": 61,
                        "objectID": "19270686",
                        "points": 705,
                        "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                        "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                      }
                    }
                  />,
                ],
                <ListChild
                  className="pagination"
                >
                  <Link
                    aria-label="click here for more news"
                    className="more-link"
                    replace={false}
                    to="?page=1"
                  >
                    More News
                  </Link>
                </ListChild>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "19270646",
                "nodeType": "class",
                "props": Object {
                  "hideNewsHandler": [Function],
                  "item": Object {
                    "author": "Vinnl",
                    "comment_text": null,
                    "created_at": "2019-02-28T11:38:16.000Z",
                    "created_at_i": 1551353896,
                    "num_comments": 61,
                    "objectID": "19270646",
                    "points": 705,
                    "title": "Mozilla releases the largest to-date public domain transcribed voice dataset",
                    "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "19270686",
                "nodeType": "class",
                "props": Object {
                  "hideNewsHandler": [Function],
                  "item": Object {
                    "author": "Vinnla",
                    "comment_text": null,
                    "created_at": "2019-02-28T11:38:16.000Z",
                    "created_at_i": 1551353896,
                    "num_comments": 61,
                    "objectID": "19270686",
                    "points": 705,
                    "title": "Mozilla releases the largest to-date public domain transcribed voice dataseta",
                    "url": "https://blog.mozilla.org/blog/2019/02/28/sharing-our-common-voices-mozilla-releases-the-largest-to-date-public-domain-transcribed-voice-dataset/",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Link
                    aria-label="click here for more news"
                    className="more-link"
                    replace={false}
                    to="?page=1"
                  >
                    More News
                  </Link>,
                  "className": "pagination",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "click here for more news",
                    "children": "More News",
                    "className": "more-link",
                    "replace": false,
                    "to": "?page=1",
                  },
                  "ref": null,
                  "rendered": "More News",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": Symbol(react.fragment),
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
